### ISSUE TITLE
try to extract layer ‘fc1’ feature using C++，but get an error

### ISSUE THREAD

**Issue by yzk0281** [AlfredXiangWu/face_verification_experiment#38]
Pretrained model:
LightenedCNN_B.caffemodel
Net proto: LightenedCNN_B_ex.prototxt but change the input to (so that I can read the image_path in commend line): 
layers {
  name: "icub-camera"
  type: MEMORY_DATA 
  top: "data"
  top: "label"
  memory_data_param {
    batch_size: 1
    channels: 1 
    height: 144
    width: 144
  }  
} 

Report
Cannot copy param 0 weights from layer 'fc1'; shape mismatch. Source param shape is 1 1 512 8192 (4194304); target param shape is 512 10368 (5308416). 

What is wrong?

**Comment by AlfredXiangWu** [comment]
> Source param shape is 1 1 512 8192 (4194304); target param shape is 512 10368 (5308416).

I think you might check your C++ code and network configuration. 
By the way, the caffemodel is trained by caffe-rc rather than caffe-rc2. Besides, the "fc1" is not the feature layer. The feature layer for network A is "eltwise6" and it for B is "eltwise_fc1".

**Comment by yzk0281** [comment]
what is caffe-rc and caffe-rc2?

**Comment by AlfredXiangWu** [comment]
There are all the caffe released versions. https://github.com/BVLC/caffe/releases

**Comment by yzk0281** [comment]
ok, I will try again, thanks~~~

**Comment by yzk0281** [comment]
我说的特征是最后生成的人脸特征向量 fc1是512维的向量

**Comment by AlfredXiangWu** [comment]
特征层对于A网络是eltwise6，对于B网络是eltwise_fc1，fc1是没有经过激活函数的，最后的特征是256维

**Comment by yzk0281** [comment]
噢。。eltwise_fc1才是最后的256维的特征阿 。。。。
另外请教一下vggface的特征层是fc7吗？

**Comment by AlfredXiangWu** [comment]
vgg的网络结构不在手边，我记得vgg是和imagenet的网络类似，都是两个全连接层，那么他的特征层应该是第二个全连接层才对

**Comment by iscas-lee** [comment]
@yzk0281  
hi，正在研究如何用wu的model对一张照片进行特征提取，能share下你的c++ or matlab 特征提取的code吗？我的email: 2759755502@qq.com

**Comment by yzk0281** [comment]
我怎么提取出eltwise_fc1还是512维的呢 。。。

**Comment by gxcLe** [comment]
hi，最近用您的模型和deploy测试了一下lfw，只有85%左右的准确率，我将图像取值范围放到0-1，特征层的提取也是按照您说的，输入的图像是灰度图，人脸矫正用的是sdm，问题会出在哪呢？期待回复，谢谢。
